src/server/pmix_server.c

```diff
@@ -289,40 +294,51 @@ PMIX_EXPORT pmix_status_t PMIx_server_init(pmix_server_module_t *module,
                     lt->mode = info[n].value.data.uint32;
                 }
             } else if (0 == strcmp(info[n].key, PMIX_SERVER_TOOL_SUPPORT)) {
-                pmix_listener_t *tl = PMIX_NEW(pmix_listener_t);
-                tl -> address.sun_family = AF_UNIX;
-                tl->protocol = PMIX_PROTOCOL_TOOL;
-                /* Get up to 30 chars of hostname.*/
-                gethostname(myhostname, myhostnamelen);
-                /* ensure it is NULL terminated */
-                myhostname[myhostnamelen-1] = '\0';
-                /* need to put this in the global tmpdir as opposed to
-                 * where the server tmpdir might be */
-                if (NULL == (tdir = getenv("TMPDIR"))) {
-                    if (NULL == (tdir = getenv("TEMP"))) {
-                        if (NULL == (tdir = getenv("TMP"))) {
-                            tdir = "/tmp";
-                        }
-                    }
-                }
-                if (0 > asprintf(&pmix_pid, "%s/pmix.%s.tool.%d", tdir, myhostname, mypid)) {
-                    return PMIX_ERR_NOMEM;
-                }
-                if ((strlen(pmix_pid) + 1) > sizeof(tl->address.sun_path)-1) {
-                    free(pmix_pid);
-                    return PMIX_ERR_INVALID_LENGTH;
+                /* defer processing to ensure we pickup any tmpdir
+                 * directives before setting location */
+                tool_support = true;
+            } else if (0 == strcmp(info[n].key, PMIX_SERVER_TMPDIR)) {
```

Comment:
**Minor comment**: Key macro has the same name as env variable. May lead to confusion at some point.